File: spigot/src/main/java/com/hibiscusmc/hmcleaves/paper/util/AdventureUtil.java
package com.hibiscusmc.hmcleaves.paper.util;

import net.kyori.adventure.text.Component;
import net.kyori.adventure.text.minimessage.MiniMessage;

public final class AdventureUtil {

    private AdventureUtil() {
        throw new UnsupportedOperationException();
    }

    public static final MiniMessage MINI_MESSAGE = MiniMessage.miniMessage();

    public static Component parse(String text) {
        return MINI_MESSAGE.deserialize(text);
    }

}

File: common/src/main/kotlin/com/hibiscusmc/hmcleaves/util/AdventureUtil.kt
package com.hibiscusmc.hmcleaves.util

import net.kyori.adventure.text.format.TextDecoration
import net.kyori.adventure.text.minimessage.MiniMessage
import net.kyori.adventure.text.minimessage.tag.standard.StandardTags
import net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer


val ADVENTURE_SERIALIZER = LegacyComponentSerializer.builder()
    .hexColors()
    .useUnusualXRepeatedCharacterHexFormat()
    .build()

val MINI_MESAGE = MiniMessage.builder()
    .tags(StandardTags.defaults())
    .postProcessor { component -> component.decoration(TextDecoration.ITALIC, false) }
    .build()
